lib/cache_mngr.c

```diff
@@ -185,23 +220,13 @@ int nl_cache_mngr_alloc(struct nl_sock *sk, int protocol, int flags,
 	if ((err = nl_socket_set_nonblocking(mngr->cm_sock)) < 0)
 		goto errout;
 
-	/* Create and allocate socket for sync cache fills */
-	mngr->cm_sync_sock = nl_socket_alloc();
-	if (!mngr->cm_sync_sock) {
-		err = -NLE_NOMEM;
-		goto errout;
-	}
-	if ((err = nl_connect(mngr->cm_sync_sock, protocol)) < 0)
-		goto errout_free_sync_sock;
```

Comment:
not how for `mngr->cm_sock`, we call connect and non-blocking above. That means, for the `sk` argument, the caller is *not* expected to do that. That may be a questionable choice, but it is probably better to handle his consistently. Meaning: leave the `nl_connect(cm_sync_sock)` call here, also for the externally provided `sk_blocking`.